moto/sesv2/urls.py

```diff
@@ -0,0 +1,14 @@
+"""sesv2 base URL and path."""
+from .responses import SESV2Response
+
+url_bases = [
+    r"https?://email\.(.+)\.amazonaws\.com",
```

Comment:
It looks like this endpoint is the same as sesv1, which means it will be difficult for Moto to figure out which service the request was made to.
From what I can tell, the only difference between them is the requestUri: 

https://github.com/boto/botocore/blob/e3375193deedb139045a19ccf821b30c2a100098/botocore/data/sesv2/2019-09-27/service-2.json#L1186

So we may have to do something similar to what we did for S3Control, which had the same issue:

https://github.com/getmoto/moto/blob/37f14567472b3b40d44b20cbba1cbd631d954f5a/moto/moto_server/werkzeug_app.py#L152

Let me know if you want me to look into this further

Reply:
Hmm.. I will need a hand with making `test_server` pass. I added an endpoint for [ListContacts](https://docs.aws.amazon.com/ses/latest/APIReference-V2/API_ListContacts.html) and I thought this will make moto server happy but not so:

```
pytest tests/test_sesv2/test_server.py                                                                                                    <aws:raf>
================================================================================ test session starts ================================================================================
platform darwin -- Python 3.9.13, pytest-7.1.3, pluggy-1.0.0
rootdir: /Users/raf/projects/moto, configfile: setup.cfg
plugins: anyio-3.6.1, mock-3.10.0, xdist-3.2.1, ordering-0.6, cov-4.0.0
collected 1 item                                                                                                                                                                    

tests/test_sesv2/test_server.py F                                                                                                                                             [100%]

===================================================================================== FAILURES ======================================================================================
__________________________________________________________________________________ test_sesv2_list __________________________________________________________________________________

    def test_sesv2_list():
        backend = server.create_backend_app("sesv2")
        test_client = backend.test_client()
    
        resp = test_client.post("/v2/email/contact-lists/test/contacts")
    
>       assert resp.status_code == 200
E       assert 404 == 200
E        +  where 404 = <WrapperTestResponse streamed [404 NOT FOUND]>.status_code

tests/test_sesv2/test_server.py:12: AssertionError
============================================================================== short test summary info ==============================================================================
FAILED tests/test_sesv2/test_server.py::test_sesv2_list - assert 404 == 200
  
```

Reply:
Ah, I don't think we need `test_server.py` actually. It's great if we want to test the raw HTTP responses, but that's not necessary in this case.

Just need to make sure that the regular boto3-tests pass in ServerMode. This was the last failure: 
https://github.com/getmoto/moto/actions/runs/4819031481/jobs/8582608793

Docs on how to run these tests locally: http://docs.getmoto.org/en/latest/docs/contributing/development_tips/tests.html#servermode-tests

Reply:
It looks like the two(my test_server test and ServerMode tests) are definitely related, they both fail for the same reason -  a 404 from the flask app 🤔 

```
tests/test_sesv2/test_sesv2.py::test_list_contacts 127.0.0.1 - - [27/Apr/2023 15:28:16] "POST /moto-api/reset HTTP/1.1" 200 -
127.0.0.1 - - [27/Apr/2023 15:28:16] "GET /v2/email/contact-lists/test/contacts HTTP/1.1" 404 -
FAILED
```